pkg/distributor/aggregator/aggregator_test.go

```diff
@@ -0,0 +1,224 @@
+package aggregator
+
+import (
+	"context"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_Aggregation(t *testing.T) {
+	w := time.Second * 15
+	d := time.Millisecond * 10
+
+	fn := func(i int) (int, error) {
```

Comment:
See using generics here, makes your testing life so much easier. Sounds like a good reason for using generics even if you don't specially need it.